internal/bminventory/inventory.go

```diff
@@ -3135,3 +3174,64 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
 	}
 	return buf.String(), nil
 }
+
+func (b bareMetalInventory) PermanentlyDeleteUnregisteredClustersAndHosts() {
+	olderThen := strfmt.DateTime(time.Now().Add(-b.Config.DeletedUnregisteredAfter))
+
+	b.log.Debugf(
+		"Permanently deleting all clusters that were de-registered before %s",
+		olderThen)
+	if err := b.permanentClustersDeletion(context.Background(), olderThen); err != nil {
+		b.log.WithError(err).Errorf("Failed deleting de-registered clusters")
+		return
+	}
+
+	b.log.Debugf(
+		"Permanently deleting all hosts that were soft-deleted before %s",
+		olderThen)
+	if err := b.permanentHostsDeletion(olderThen); err != nil {
+		b.log.WithError(err).Errorf("Failed deleting soft-deleted hosts")
+		return
+	}
+}
+
+func (b bareMetalInventory) permanentClustersDeletion(ctx context.Context, olderThen strfmt.DateTime) error {
+	var clusters []*common.Cluster
+	db := b.db.Unscoped()
+	if reply := db.Where("deleted_at < ?", olderThen).Find(&clusters); reply.Error != nil {
+		return reply.Error
+	}
+	for _, c := range clusters {
+		if err := b.deleteAllS3ClusterFiles(ctx, c); err != nil {
+			return err
+		}
+		if reply := db.Delete(&c); reply.Error != nil {
+			return reply.Error
+		} else if reply.RowsAffected > 0 {
+			b.log.Debugf("Deleted %s cluster from db", reply.RowsAffected)
+		}
+	}
+	return nil
+}
+
+func (b bareMetalInventory) deleteAllS3ClusterFiles(ctx context.Context, c *common.Cluster) error {
```

Comment:
Ask @tsorya amd @ybettan if this is a good way to handle logs retention, or should it be done using S3?

Reply:
sounds good for me, why not?